src/js/actions/layers.js

```diff
@@ -1239,6 +1239,30 @@ define(function (require, exports) {
     setVisibility.writes = [locks.PS_DOC, locks.JS_DOC];
 
     /**
+     * Used by the show/hide menu items, targets the active document active layers
+     *
+     * @param {boolean} visible
+     * @returns {Promise}
+     */
+    var setVisibilitySelectedInCurrentDocument = function (visible) {
+        var applicationStore = this.flux.store("application"),
+            currentDocument = applicationStore.getCurrentDocument();
+
+        if (!currentDocument) {
+            return Promise.resolve();
+        }
+
+        var showPromises = currentDocument.layers.selected.map(function (layer) {
+            return this.transfer(setVisibility, currentDocument, layer, visible);
+        }, this).toArray();
+
+        return Promise.all(showPromises);
+    };
+    setVisibilitySelectedInCurrentDocument.reads = [locks.JS_DOC, locks.PS_DOC];
```

Comment:
you're not actually reading the PS_DOC, right?   I know it doesn't actually matter here because of the transfer, but asking anyway.